vscode-lean4/test/README.md

```diff
@@ -0,0 +1,65 @@
+install lives in a local `.vscode-test` folder, and it has it's own user data in there also. So the
+first time you run the tests you will see it download the latest version of VS code into that
+folder.  It launches this test version with a `--extensionDevelopmentPath` option pointing to the
+`vscode-lean4` folder so that the local version of the extension you have build loads into VS code
+for testing.  All this is setup by the [runTests.ts](src/runTests.ts) program.
+
+The test folder is organized into:
+- **src** - for the actual test code.
+- **suite** - contains Lean sample projects used by the tests.
+
+The following is a description of the tests:
+- **src/lean3** - tests that the lean3 extension loads and the lean4 extension does not get in the way.
+- **src/simple** - contains 3 tests:
+  - `Untitled Lean File` tests that the lean4 extension loads correctly for adhoc files, and untitled files, and that the infoview opens and contains the right output, so this is an end-to-end test ensuring the lean language service is running.  It also tests that you can `Goto Definition` to the Lean source code for `Lean.versionString` and that this takes you to `leanprover--lean4---nightly`.
+  - `Orphaned Lean File` tests we can open a Lean 4 file in a folder that has no inherited `lean-toolchain` version information and that you get the `default` toolchain in this case.
+  - `Goto definition in a package folder` tests opening a folder containing a Lean 4 project, and that goto definition works across files in the project.
+  - `Restart Server` tests the `Lean4: Restart Server` command.
+- **src/multi** - contains 3 tests:
+  - `Load a multi-project workspace` tests lean4 works in a multi-folder VS code workspace where each folder in the workspace uses a different version of lean.  It verifies that 2 separate LeanClients are running in this case.
+  - `Select toolchain` tests the `Lean4: Select Toolchain` command can override the lean version used in one of those projects and that this override can be reset.
+  - `Edit lean-toolchain version` tests that when you edit the `lean-toolchain` file and specify a
+  different version that the lean server is restarted with that new version.
+
+## Running the tests
+
+To run the tests you must first follow the build steps in [../readme.md](../readme.md).
+
+For these tests to pass on your machine you need to:
+- install the lean3 extension
+- install `leanprover/lean4:nightly` toolchain
+- install `leanprover/lean4:stable` toolchain
+- ensure you have a `default` Lean4 toolchain
+- create a linked toolchain named `master` that points to a build of lean4 bits on your machine.
+See [on-push.yml](../../.github/workflows/on-push.yml) for example of how to set that up).
+
+Now you can run `npx lerna run test` or if your terminal window is inside `vstest-lean4` package
+folder you can run `npm run test` and you will get some handy console output of the test run.
```

Comment:
With the suggested change to the top-level `package.json`:
```suggestion
Now you can run `npm run test` and you will get some handy console output of the test run.
```

Reply:
Actually, npm run test at the root gives me this:
```
(ell) d:\git\leanprover\vscode-lean4>npm run test
npm ERR! Missing script: "test"
npm ERR!
npm ERR! To see a list of scripts, run:
npm ERR!   npm run

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\clovett\AppData\Local\npm-cache\_logs\2022-03-01T04_09_47_503Z-debug-0.log

```

Reply:
Well, you'd need to apply the other suggested change first!

Reply:
got it, thanks.